src/renderer/decoder/lineDecoder.js

```diff
@@ -4,29 +4,30 @@ import { getFloat32ArrayFromArray } from '../../utils/util';
 // If the geometry type is 'line' it will generate the appropriate zero-sized, vertex-shader expanded triangle list with `miter` and `bevel` joins.
 // The geom will be an array of coordinates in this case
 
+let vertices = new Array(1000000);
+let normals = new Array(1000000);
+
 export function decodeLine (geometry) {
-    let vertices = [];
-    let normals = [];
     let breakpoints = []; // Array of indices (to vertexArray) that separate each feature
     let featureIDToVertexIndex = new Map();
+    let vertexIndex = 0;
 
     for (let i = 0; i < geometry.length; i++) {
         const feature = geometry[i];
         for (let j = 0; j < feature.length; j++) {
-            const lineString = feature[j];
-            addLine(lineString, vertices, normals);
+            vertexIndex = addLine(feature[j], vertices, normals, vertexIndex);
         }
 
         featureIDToVertexIndex.set(breakpoints.length, breakpoints.length === 0
-            ? { start: 0, end: vertices.length }
-            : { start: featureIDToVertexIndex.get(breakpoints.length - 1).end, end: vertices.length });
+            ? { start: 0, end: vertexIndex }
+            : { start: featureIDToVertexIndex.get(breakpoints.length - 1).end, end: vertexIndex });
 
-        breakpoints.push(vertices.length);
+        breakpoints.push(vertexIndex);
     }
 
     return {
-        vertices: getFloat32ArrayFromArray(vertices),
-        normals: getFloat32ArrayFromArray(normals),
+        vertices: getFloat32ArrayFromArray(vertices, vertexIndex),
```

Comment:
Since `vertices` must be copied to a `Float32Array` anyway having the static buffers seems like a really good approach :rocket: